Language/Functions/element_type_A02_t03.dart

```diff
@@ -27,6 +28,7 @@ void isRuntimeTypeImplementsStream<T>(Object? o) async {
   List<T> list = await o.toList();
   try {
     list.addAll(<T>[]);
+    asyncEnd();
```

Comment:
Same as my other comment, I think this should be the last line in `main`.

Reply:
Revisiting this after my other comment, it looks like this `asyncEnd` is correct, because the part after the `await` in this function will run after `main` ends.

It would be helpful if someone else could confirm (maybe @lrhn or @mkustermann).

Reply:
It's a correct change, but I'd write it differently (imho) somewhat more cleanly like this:
```dart
void isRuntimeTypeImplementsStream<T>(Object? o) async {
  ...
  asyncStart();
  List<T> list = await o.toList();
  try {
    list.addAll(<T>[]);
    asyncEnd();
  } catch (...) {
    ...
  }
}

main() async {
  asyncStart();
  dynamic d = await foo();
  FutureOr<Stream<int>> o = d;
  isRuntimeTypeImplementsStream<int>(d);
  asyncEnd();
}
```

This works because we're going to call `asyncStart()` before any async operation. So the counter will go up to 2, then `main` decreases it and then finally `isRuntimeTypeImplementsStream` decreases it and reports success.

The code is easier to read as the `asyncStart` and `asyncEnd` are contained within functions and balanced.